### DISCUSSION TITLE
MIRTH CLIENT API - Getting only one message

### DISCUSSION THREAD

**Discussion by boukenka** [nextgenhealthcare/connect#5219] [Q&A] [answered]
Hello,

I am doing the following request : https://localhost:8444/api/channels/5662c269-12d9-4a7f-af96-d9dc94946434/messages?textSearchRegex=false&includeContent=false&limit=20&offset=0

It returns only one message. But there are several messages. What am I missing?

![image](https://user-images.githubusercontent.com/16502423/171200900-d6c9b177-6039-4e9b-9279-c6e1e5e1c725.png)

Thanks for any answer.

**Comment by siddharthsharma9258** [comment]
I think it is because you are using both limit and offset in your API call. Try removing offset and see if changes the results.

**Comment by boukenka** [comment, marked as answer]
@siddharthsharma9258 If I remove the _offset_ criteria, the answer is a ```<java.lang.NullPointerException>``` and the status code is 500. The same goes if I keep the _offset_ criteria and I remove the _limit_ criteria.

**Reply by boukenka** [reply]
I could finally get all the messages. Thank you for your help @siddharthsharma9258

**Comment by siddharthsharma9258** [comment]
What was the problem?

**Reply by boukenka** [reply]
It was a matter of channel selection.
